src/qml/DigitizingToolbar.qml

```diff
@@ -242,7 +242,19 @@ QfVisibilityFadingRow {
       if (Number(rubberbandModel.geometryType) === Qgis.GeometryType.Point || Number(rubberbandModel.geometryType) === Qgis.GeometryType.Null) {
         confirm();
       } else {
-        addVertex();
+        if (settings.valueBool("/QField/Digitizing/CurveEdition", false) == true) {
```

Comment:
I think we need an additional check here to avoid curved edition when people are freehand drawing, otherwise worlds will collide.